cpp/src/arrow/flight/sql/odbc/flight_sql/config/configuration.cc

```diff
@@ -155,7 +153,14 @@ const std::string& Configuration::Get(const std::string_view& key) const {
 void Configuration::Set(const std::string_view& key, const std::string& value) {
   const std::string copy = boost::trim_copy(value);
   if (!copy.empty()) {
-    this->properties[key] = value;
+    this->properties[std::string(key)] = value;
```

Comment:
Shouldn't this be properties[copy]?

Reply:
`copy` holds a copy of the `value`, we still need to construct a `std::string` copy for `key`